Add modal issue

praffi-systemx opened the pull request:
- add function to stop computation if any alarm is raised
- add modal triggered when any alarm is raised
- fix logger level and add --logging_level to app arguments
- add --test_alarms as temporary test mode to force all alarms to trigger at any step

praffi-systemx committed:
Add modal issue

- add function to stop computation if any alarm is raised
- add modal triggered when any alarm is raised
- fix logger level and add --logging_level to app arguments

praffi-systemx committed:
Add recommandations table

BDonnot commented:
Let me know when you'll "freeze" the feature on this pull request so that I can have a look 

Thanks

praffi-systemx commented:
You can have a look, I'm freezing this pull request and creating a new branch from dev_cab for improvement and other features.

praffi-systemx committed:
Add to variant trees

praffi-systemx committed:
Add running modes

BDonnot commented:
I did not look at the code yet, but from a user perspective:

- it would be nice to differentiate between the "legacy" grid2game mode and the one developed by CAB project, maybe with a different command line argument ? Or another option in the command line argument ? I think it could be done without too much trouble and it would be easier to differentiate between the different usecase of the application.
- the tab, when there is an issue, is "locked" we cannot do anything with it yet. I'm not sure it is ready to be merged on `main` yet
- the "overload" column of this tab should also write the id (or better the name) of the line on overflow no ? (maybe see with Alexandre or Antoine)
- to help debugging, I think it would be better to write the interaction in the logger at least so we see that the "clicks" on the button are properly recorded. For now i'm not sure if it's a "backend" issue (*ie* the clicks are properly recorded by nothing happens) or if it's a "front" issue (clicks are not detected correctly) or if it's something in between (*eg* no callbacks)

I'll look at the code asap to give more technical feedback if needed

praffi-systemx commented:
Thank you for the review

- I will add a legacy mode to the new mode selection dropdown (which is being implemented).
- It has already be planned with CAB and RTE to have the line and/or the region of the overloads, but it isn't a priority.
- The buttons in the recommendations table aren't implemented in this pull request.
- I will add some logs on the python DEBUG level to help debugging

praffi-systemx committed:
Apply recommandation

praffi-systemx committed:
Add simulate and integrate manual action

- only layout, no callbacks yet

praffi-systemx committed:
Add simulate and integrate manual action callbacks

- Simulate and integrate manual action buttons will show only if the recommendations are open.
- Simulate button doesn't work. There is a bug on main branch, simulated figure doesn't update on manual action.
- Refactoring of recommendations
- TODO: Handle multi-recommendations for human integrated actions

praffi-systemx committed:
Add simulate recommendation button

praffi-systemx committed:
Add Expert Assistance

- add expert assist tab with basic view from grid2viz/src/simulation/ExpertAssist.py
  this tab interacts with the recommendations
- add some user input persistence through tab switch
- prevent initial call on most of CAB's callbacks to fix bugs
